src/Template/User/settings.ctp

```diff
@@ -172,21 +172,21 @@ $this->set('title_for_layout', $this->Pages->formatTitle(__('Settings')));
             </md-list-item>
 
             <md-list-item>
-                <?php $hideRandomSentence = $userSettings->settings['hide_random_sentence']; ?>
+                <?php $showRandomSentence = 1 - $userSettings->settings['hide_random_sentence']; ?>
                 <md-checkbox
                     ng-false-value="0"
                     ng-true-value="1"
-                    ng-model="hideRandomSentence"
-                    ng-init="hideRandomSentence = <?= $hideRandomSentence ?>"
+                    ng-model="showRandomSentence"
+                    ng-init="showRandomSentence = <?= $showRandomSentence ?>"
```

Comment:
You can instead just swap the `ng-false-value` and `ng-true-value`.

`ng-false-value="0"` means that when the checkbox is unchecked, the value for the binded variable is 0.

Reply:
I thought so but the default value of `hide_random_sentence` being `false`, I couldn't make it work for new users. The false is not interpreted as 0 so the html is `ng-init="showRandomSentence = "` and the checkbox is unchecked. I used `1 - false` to force it to `1` and have it work as expected. Am I missing something? 

Should we set `hide_random_sentence` to `'0'` instead of `false` by default?

Reply:
The `ng-init` line doesn't work because a PHP false value is casted to the empty string when printing/echoing it.

If we want to keep 1 as true and 0 as false on the JS side we should use these values also on the PHP side for all the boolean settings.

Reply:
Yea, I remember this issue from redesigning the advanced search. I did this for the sort checkbox:
```
ng-init="sortReverse = <?= $sort_reverse == 'yes' ? 1 : 0 ?>"
```

https://github.com/Tatoeba/tatoeba2/blob/dev/src/Template/Element/advanced_search_form.ctp#L323